File: Algorithms/Random/Index-with-Equal-Left-and-Right-Sums.py
class Solution:
    def solve(self, nums):

        if sum(nums)-nums[0] == 0:
            return 0

        prefix, suffix = 0, sum(nums)

        for i in range(len(nums)):
            suffix -= nums[i]
            if prefix == suffix:
                return i
            prefix += nums[i]
        return -1

File: C++/Index-with-Equal-Left-and-Right-Sums.cpp
int solve(vector<int>& nums) {
    int sum = 0;

    for (int i = 0; i<nums.size(); i++) {
        sum+=nums[i];
    }

    sum -= nums[0];

    if (sum == 0) {
        return 0;
    }

    int prefix, suffix;
    
    prefix = 0;
    suffix = sum+nums[0];

    for (int i = 0; i<nums.size(); i++) {
        suffix-=nums[i];
        if (prefix == suffix) {
            return i;
        }
        prefix+=nums[i];
    }

    return -1;

}
